azurelinuxagent/common/osutil/nsbsd.py

```diff
@@ -0,0 +1,142 @@
+#
+# Copyright 2018 Stormshield
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import azurelinuxagent.common.utils.fileutil as fileutil
+import azurelinuxagent.common.utils.shellutil as shellutil
+import azurelinuxagent.common.utils.textutil as textutil
+import azurelinuxagent.common.logger as logger
+from azurelinuxagent.common.exception import OSUtilError
+from azurelinuxagent.common.osutil.freebsd import FreeBSDOSUtil
+from azurelinuxagent.common.future import ustr
+import azurelinuxagent.common.conf as conf
+import os
+import time
+
+class NSBSDOSUtil(FreeBSDOSUtil):
+
+    resolver = None
+
+    def __init__(self):
+        super(NSBSDOSUtil, self).__init__()
+
+        if self.resolver is None:
+            # NSBSD doesn't have a system resolver, configure a python one
+
+            try:
+                import dns.resolver
+            except ImportError:
+                logger.warn("Python DNS resolver not available")
+                return
```

Comment:
Isn't this more of a fatal error rather than just a warning? There's no system resolver in NSBSD, you can't load a python resolver... that's going to make hostname resolution pretty difficult. Better to fail with a very clear error message, generating the appropriate telemetry traces.

Reply:
ok, I'll patch that